packages/react/src/use-feature-flag.ts

```diff
@@ -0,0 +1,47 @@
+import { Client, EvaluationDetails, FlagValue, ProviderEvents, ResolutionDetails } from '@openfeature/web-sdk';
+import { useEffect, useState } from 'react';
+import { useOpenFeatureClient } from './provider';
+
+export function useFeatureFlag<T extends FlagValue>(flagKey: string, defaultValue: T): T {
+  const [flagEvaluationDetails, setFlagEvaluationDetails] = useState<ResolutionDetails<T> | null>(null);
+
+  const client = useOpenFeatureClient();
+
+  useEffect(() => {
+    getFlag(client, flagKey, defaultValue, setFlagEvaluationDetails);
+    // adding handlers here mean that changes are immediately reflected in the UI when flags change
+    client.addHandler(ProviderEvents.Ready, () => {
+      getFlag(client, flagKey, defaultValue, setFlagEvaluationDetails);
+    });
+    client.addHandler(ProviderEvents.ConfigurationChanged, () => {
+      getFlag(client, flagKey, defaultValue, setFlagEvaluationDetails);
+    });
```

Comment:
Feels like this should return a cleanup function that removes these handlers.

Reply:
absolutely.